test/utils/runCodeServerCommand.ts

```diff
@@ -0,0 +1,86 @@
+import { spawn } from "child_process"
+
+type RunCodeServerCommandOptions = {
+  stderr?: "log"
+  stdout?: "log"
+  ignoreFail?: boolean
+}
+
+interface ErrorWithMoreInfo extends Error {
+  stderr: string
+  stdout: string
+}
+
+type CSCmd =
+  | {
+      code: number | null
+      signal: NodeJS.Signals | null
+      stdout: string
+      stderr: string
+    }
+  | ErrorWithMoreInfo
+
+/**
+ *
+ * A helper function for integration tests to run code-server commands.
+ */
+export async function runCodeServerCommand(argv: string[], options: RunCodeServerCommandOptions = {}): Promise<CSCmd> {
```

Comment:
We could use `exec` (either directly in the tests or with a thin wrapper here) which I think does everything this function does (it even has stdout/stderr on thrown errors):

```js
const { stdout, stderr } = await util.promisify(cp.exec)(CODE_SERVER_PATH, argv)
```

Reply:
Yeah, you could keep this exact same function but just replace all the contents with `return util.promisify(cp.exec)(CODE_SERVER_PATH, argv)`.  Just seems unfortunate to reimplement a builtin! :)

Reply:
Note to self, i think `argv` must be a string: `exec('"/path/to/test file/test.sh" arg1 arg2');`

```typescript
  const { stdout, stderr } = await promisify(exec)(`${CODE_SERVER_COMMAND} ${argv.join(" ")}`)
```

Reply:
Ohhh yeah you are right it does not accept an array.  :facepalm:

Reply:
Dude, I just removed so much code. I went with the thin wrapper approach. Nice!